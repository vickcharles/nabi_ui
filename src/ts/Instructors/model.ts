export interface ExperienceType {
  id?: number;
  employer: string;
  jobTitle: string;
  fromMonth: string;
  fromYear: string;
  toMonth: string;
  toYear: string;
  location: string;
}

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degreee',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface EducationType {
  id?: number;
  schoolName: string;
  graduationYear: string;
  degreeType: DegreeType;
  fieldOfStudy: string;
}

export enum PlaceForLessonsOptions {
  home = 'Student\'s home',
  studio = 'Instructor\'s studio',
  online = 'Online',
}

export interface PlaceForLessonsType {
  home: boolean;
  studio: boolean;
  online: boolean;
}

export enum LessonDuration {
  thirtyMins = '30 mins',
  fortyFiveMins = '45 mins',
  sixtyMins = '60 mins',
  ninetyMins = '90 mins'
}

export interface RatesType {
  thirtyMinsRate: number;
  fortyFiveMinsRate: number;
  sixtyMinsRate: number;
  ninetyMinsRate: number;
}

export enum SkillLevel {
  beginner = 'Beginner',
  intermediate = 'Intermediate',
  advanced = 'Advanced'
}

export interface InstrumentsType {
  instrument: string;
  skillLevel: SkillLevel;
}

export interface InstructorType {
  userId: string;
  bio?: string;
  instruments?: InstrumentsType[];
  rates?: RatesType;
  placeForLessons?: PlaceForLessonsType;
  studioAddress?: string;
  education?: EducationType[];
  experience?: ExperienceType[];
}

export type IState = InstructorType[];
