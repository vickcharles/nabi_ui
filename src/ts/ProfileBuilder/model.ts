export interface InstructorState {
  userId: string;
  bio?: string;
}

export type IState = InstructorState[];
  