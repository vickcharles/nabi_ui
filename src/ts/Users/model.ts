export enum HearAboutUs {
  Google = 'google',
  Facebook = 'facebook',
  Craigslist = 'craigslist',
  PrintAds = 'printAds',
  WordOfMouth = 'wordOfMouth',
  CurrentNabiUser = 'currentNabiUser',
  Other= 'other'
}

export interface HearAboutUsInfo {
  value: HearAboutUs;
  label: string;
}

export const hearAboutUsInfo = {
  Google: <HearAboutUsInfo> {
    value: HearAboutUs.Google,
    label: 'Google'
  },
  Facebook: <HearAboutUsInfo> {
    value: HearAboutUs.Facebook,
    label: 'Facebook'
  },
  Craigslist: <HearAboutUsInfo> {
    value: HearAboutUs.Craigslist,
    label: 'Craigslist'
  },
  PrintAds: <HearAboutUsInfo> {
    value: HearAboutUs.PrintAds,
    label: 'Print Ads'
  },
  WordOfMouth: <HearAboutUsInfo> {
    value: HearAboutUs.WordOfMouth,
    label: 'Word Of Mouth'
  },
  CurrentNabiUser: <HearAboutUsInfo> {
    value: HearAboutUs.CurrentNabiUser,
    label: 'Current Nabi User'
  },
  Other: <HearAboutUsInfo> {
    value: HearAboutUs.Other,
    label: 'Other'
  }
};

export enum Role {
  student = 'student',
  instructor = 'instructor',
  parent = 'parent'
}

export interface UserState {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  zipCode: string;
  role: Role;
  hearAboutUs: string;
  avatar?: string;
  displayName: string;
}

export type IState = UserState[];
