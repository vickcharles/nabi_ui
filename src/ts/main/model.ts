/**
 * State for firing redirect
 * @interface RedirectState
 */
export interface RedirectState {
  fireRedirect: boolean;
}
