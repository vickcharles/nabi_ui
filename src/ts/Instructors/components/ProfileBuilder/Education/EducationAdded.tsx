import * as React from 'react';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import Typography from 'material-ui/Typography';
import { EducationType }  from '../../../model';
import Divider from 'material-ui/Divider';

/**
 * Props for EducationAdded
 * @interface EducationAddedProps
 */
interface EducationAddedProps extends EducationType {
  deleteEducation: (educationId: any) => void;
}

/**
 * EducationAdded
 * @extends React.StatelessComponent<EducationAddedProps>
 */
const EducationAdded: React.StatelessComponent<EducationAddedProps> = props => {
  return (
    <div>
      <div className="selected-education">
        <IconButton className="nabi-float-right" aria-label="Delete" onClick={() => props.deleteEducation(props.id)}>
          <DeleteIcon />
        </IconButton>
        <Typography className="nabi-margin-top-small" variant="body2">
          {props.schoolName.toUpperCase()}
        </Typography>
        <Typography className="nabi-margin-top-small">
          {`${props.degreeType}, ${props.fieldOfStudy}`}
        </Typography>
        <Typography className="nabi-margin-top-small">
          {props.graduationYear}
        </Typography>
      </div>
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default EducationAdded;
