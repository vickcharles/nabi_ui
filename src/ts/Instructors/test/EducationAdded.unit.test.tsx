import * as React from 'react';
import EducationAdded from '../components/ProfileBuilder/Education/EducationAdded';
import { shallow } from 'enzyme';
import { DegreeType } from '../model';

describe('EducationAdded', () => {
  let wrapper: any;
  const deleteEducation: () => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <EducationAdded
          deleteEducation={deleteEducation}
          schoolName=""
          graduationYear=""
          degreeType={DegreeType.bachelors}
          fieldOfStudy=""
        />
      );
    });
    
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
