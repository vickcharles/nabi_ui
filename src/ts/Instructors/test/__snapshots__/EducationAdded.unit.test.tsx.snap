// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationAdded Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EducationAdded
    degreeType="Bachelor's Degree"
    deleteEducation={[MockFunction]}
    fieldOfStudy=""
    graduationYear=""
    schoolName=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="selected-education"
        >
          <WithStyles(IconButton)
            aria-label="Delete"
            className="nabi-float-right"
            onClick={[Function]}
          >
            <pure(Delete) />
          </WithStyles(IconButton)>
          <WithStyles(Typography)
            className="nabi-margin-top-small"
            variant="body2"
          >
            
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className="nabi-margin-top-small"
          >
            Bachelor's Degree, 
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className="nabi-margin-top-small"
          >
            
          </WithStyles(Typography)>
        </div>,
        <WithStyles(Divider)
          className="nabi-margin-top-small"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>,
            <WithStyles(Typography)
              className="nabi-margin-top-small"
              variant="body2"
            >
              
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className="nabi-margin-top-small"
            >
              Bachelor's Degree, 
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              className="nabi-margin-top-small"
            >
              
            </WithStyles(Typography)>,
          ],
          "className": "selected-education",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <pure(Delete) />,
              "className": "nabi-float-right",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "className": "nabi-margin-top-small",
              "variant": "body2",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Bachelor's Degree, ",
              "className": "nabi-margin-top-small",
            },
            "ref": null,
            "rendered": "Bachelor's Degree, ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "className": "nabi-margin-top-small",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "nabi-margin-top-small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="selected-education"
          >
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right"
              onClick={[Function]}
            >
              <pure(Delete) />
            </WithStyles(IconButton)>
            <WithStyles(Typography)
              className="nabi-margin-top-small"
              variant="body2"
            >
              
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className="nabi-margin-top-small"
            >
              Bachelor's Degree, 
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className="nabi-margin-top-small"
            >
              
            </WithStyles(Typography)>
          </div>,
          <WithStyles(Divider)
            className="nabi-margin-top-small"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Delete"
                className="nabi-float-right"
                onClick={[Function]}
              >
                <pure(Delete) />
              </WithStyles(IconButton)>,
              <WithStyles(Typography)
                className="nabi-margin-top-small"
                variant="body2"
              >
                
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                className="nabi-margin-top-small"
              >
                Bachelor's Degree, 
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                className="nabi-margin-top-small"
              >
                
              </WithStyles(Typography)>,
            ],
            "className": "selected-education",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <pure(Delete) />,
                "className": "nabi-float-right",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "",
                "className": "nabi-margin-top-small",
                "variant": "body2",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Bachelor's Degree, ",
                "className": "nabi-margin-top-small",
              },
              "ref": null,
              "rendered": "Bachelor's Degree, ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "",
                "className": "nabi-margin-top-small",
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "nabi-margin-top-small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
